src/reducers/ui.js

```diff
@@ -18,6 +18,8 @@ const defaultState = new Immutable.Map().
     typing: false,
     requestedFocusedLine: null,
     textSizeIsLarge: false,
+    focusedSelector: null,
+    focusedEditor: null,
```

Comment:
Do we actually need both of these pieces of information? From a standpoint of application state, it seems like all we _really_ need to know is the `focusedSelector`—we don’t care which editor the selector came from. It does make sense to have separate *actions* that represent changes to editor focus and cursor position, but it’s not clear that the state itself needs to capture both pieces of information (e.g. we can just clear `focusedSelector` when the editor is blurred).

Reply:
I don't think so. There was a reason why I did it this way but it's not coming to mind. Will look into it.